pkgs/build-support/binary-cache.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, buildPackages }:
+
+# This function is for creating a flat-file binary cache, i.e. the kind created by
+# nix copy --to file:///some/path and usable as a substituter (with the file:// prefix).
+
+# For example, in the Nixpkgs repo:
+# nix-build -E 'with import ./. {}; mkBinaryCache { rootPaths = [hello]; }'
+
+{ name ? "binary-cache"
+, rootPaths
+}:
+
+stdenv.mkDerivation {
+  inherit name;
+
+  __structuredAttrs = true;
+
+  exportReferencesGraph.closure = rootPaths;
+
+  preferLocalBuild = true;
+
+  PATH = "${buildPackages.coreutils}/bin:${buildPackages.jq}/bin:${buildPackages.python3}/bin:${buildPackages.nix}/bin:${buildPackages.xz}/bin";
+
+  builder = builtins.toFile "builder" ''
+    . .attrs.sh
+
+    export out=''${outputs[out]}
+
+    mkdir $out
+    mkdir $out/nar
+
+    python <<EOF
```

Comment:
Couldn't this python script be put in a separate `.py` file in the same directory as `default.nix`, rather than being a here-doc inside of a multiline string?

Reply:
Certainly, but that would end up creating an extra store path to hold the Python file, right? I usually try to avoid creating extraneous store paths.

Reply:
Store paths are cheap.

Reply:
Please put it in a separate file.

Reply:
Done